From 200b6298365b61b76c53778a075522e1e9bd994f Mon Sep 17 00:00:00 2001
From: MSe <mse1969@posteo.de>
Date: Tue, 22 Aug 2023 20:55:55 +0200
Subject: [PATCH] Harden/improve GPS configuration

 * Switch gpsOneXtra to IZatCloud (invalid certificate)
 * Enable HTTPS (IZatCloud supports HTTPS)
 * Disable LPP (privacy)
 * Capabilities - Disable MSA (privacy) and geofencing/ULP
 * CVE-2018-9526:
   See: https://android.googlesource.com/device/google/marlin/+/fa7f7382e8b39f7ca209824f97788ab25c44f6a3

Change-Id: Iaadb4ee1bfabfb8e8aebfe95d472d9ffddc785e1
---
 etc/gps.conf | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/etc/gps.conf b/etc/gps.conf
index 3bec29a..4e3d30e 100644
--- a/etc/gps.conf
+++ b/etc/gps.conf
@@ -1,8 +1,8 @@
 #Uncommenting these urls would only enable
 #the power up auto injection and force injection(test case).
-#XTRA_SERVER_1=http://xtrapath1.izatcloud.net/xtra2.bin
-#XTRA_SERVER_2=http://xtrapath2.izatcloud.net/xtra2.bin
-#XTRA_SERVER_3=http://xtrapath3.izatcloud.net/xtra2.bin
+XTRA_SERVER_1=https://xtrapath4.izatcloud.net/xtra2.bin
+XTRA_SERVER_2=https://xtrapath5.izatcloud.net/xtra2.bin
+XTRA_SERVER_3=https://xtrapath6.izatcloud.net/xtra2.bin
 
 #Version check for XTRA
 #DISABLE = 0
@@ -17,7 +17,7 @@ XTRA_VERSION_CHECK=0
 ERR_ESTIMATE=0
 
 #Test
-NTP_SERVER=time.gpsonextra.net
+NTP_SERVER=time.izatcloud.net
 #Asia
 # NTP_SERVER=asia.pool.ntp.org
 #Europe
@@ -59,7 +59,7 @@ USE_EMERGENCY_PDN_FOR_EMERGENCY_SUPL=1
 #config.xml.
 #MSA=0X2
 #MSB=0X1
-#SUPL_MODE=
+SUPL_MODE=1
 
 # GPS Capabilities bit mask
 # SCHEDULING = 0x01
@@ -68,7 +68,7 @@ USE_EMERGENCY_PDN_FOR_EMERGENCY_SUPL=1
 # ON_DEMAND_TIME = 0x10
 # GEOFENCE = 0x20
 # default = ON_DEMAND_TIME | MSA | MSB | SCHEDULING | GEOFENCE
-CAPABILITIES=0x37
+CAPABILITIES=0x13
 
 # Accuracy threshold for intermediate positions
 # less accurate positions are ignored, 0 for passing all positions
-- 
2.39.2

